lms/djangoapps/courseware/tests/test_entrance_exam.py

```diff
@@ -206,6 +211,100 @@ def setUp(self):
             ]
         )
 
+    @mock.patch('xmodule.x_module.XModuleMixin.has_dynamic_children', mock.Mock(return_value='True'))
+    def test_view_redirect_if_entrance_exam_required(self):
```

Comment:
I directly hit the URL for a section outside of the entrance exam.

As far as the score, if you go to http://studio.pr6997.m.sandbox.edx.org/settings/details/wew/wefe/wewe you will see that the exam score should be 80% at the moment, not 100%. I went back and forth a couple times before I wrote up the bug, so the current state does not match the repro steps.